cardano-testnet/src/Testnet/EpochStateProcessing.hs

```diff
@@ -75,3 +83,35 @@ maybeExtractGovernanceActionIndex sbe txid (AnyNewEpochState actualEra newEpochS
     compareWithTxId (Api.TxId ti1) Nothing (GovActionId (TxId ti2) (L.GovActionIx gai)) _
       | ti1 == L.extractHash ti2 = Just gai
     compareWithTxId _ x _ _ = x
+
+-- | Watch the epoch state view until the guard function returns @Just@ or the timeout is reached.
+-- | Wait for at least `minWait` epochs and at most `maxWait` epochs.
+-- | The function will return the result of the guard function if it is met, otherwise it will return @Nothing@.
+watchEpochStateView
+  :: forall m a. (MonadIO m, MonadTest m, MonadAssertion m)
+  => EpochStateView -- ^ The info to access the epoch state
+  -> (AnyNewEpochState -> m (Maybe a)) -- ^ The guard function (@Just@ if the condition is met, @Nothing@ otherwise)
+  -> Word64 -- ^ The minimum number of epochs to wait
+  -> Word64 -- ^ The maximum number of epochs to wait
```

Comment:
```suggestion
  -> EpochInterval -- ^ The minimum number of epochs to wait
  -> EpochInterval -- ^ The maximum number of epochs to wait
```

Reply:
Fixed in: https://github.com/IntersectMBO/cardano-node/pull/5830/commits/a3b7e39d18dcbe5e345fc2de8e721927e89abb86